js/supermarkets.js

```diff
@@ -148,76 +174,21 @@ function setOnClickListener() {
     });
 }
 
-// function searchstores(data) {
-//     // let foundStores = [];
-//     let inputValue = document.getElementById('inputValue').value;
-//     // console.log(inputValue);
-//     for (let i = 0; i < data.markets.length; i++) {
-//     }
-// };
-
-    function searchByPlace()  {
-        //var foundStores = [];
-        //Creating the searchbox
-        let input = document.getElementById('inputValue');
-        let autocomplete = new google.maps.places.Autocomplete(input);
-        autocomplete.bindTo('bounds', map)
-        // Listen for the event fired when the user selects a prediction and retrieve
-        // more details for that place.
-          autocomplete.addListener("places_changed", function () {
-            //infoWindow.close();
-            processButtonSearch();
-            marker.setVisible(false);
-            let place = autocomplete.getPlaces();
-            if (!place.geometry) {
-                window.alert("Autocomplete's rturned place contains no geomtery");
-                return;
-            }
-            //if the place has a geometry,then present it on a map
-            if(place.geometry.viewport) {
-                map.fitBounds(place.geometry.viewport);
-            } else {
-                map.setCenter(place.geometry.location);
-                map.setZoom(17);
-            }
-          });
-
-          showStoresMarker(data);
-          displayStores(data);
-    };
-
-     processButtonSearch();
-
-    function processButtonSearch() {
-        const geocoder = new google.maps.Geocoder();
-        let button = document.getElementById('searchbutton');
-        button.addEventListener('click', function() {
-            geocodeAddress(geocoder, map);
-          });       
-    };
-    
-    function geocodeAddress(geocoder, resultsMap) {
-        var address = document.getElementById('inputValue').value;
-        geocoder.geocode({'address': address}, function(results, status) {
-          if (status === 'OK') {
-            resultsMap.setCenter(results[0].geometry.location);
-            // new google.maps.Marker({
-            //   map: resultsMap,
-            //   position: results[0].geometry.location
-            // });
-          } else {
-            alert('Geocode was not successful for the following reason: ' + status);
-          }
-        });
-      }
+function autoCompleteInSearch()  {
+    let input = document.getElementById('inputValue');
+    let autocomplete = new google.maps.places.Autocomplete(input);
+        autocomplete.bindTo('bounds', map);
+};
 
-    function clearLocations() {
-        infoWindow.close();
-        for (var i = 0; i < markers.length; i++) {
-          markers[i].setMap(null);
-        }
-        markers.length = 0;
+function clearLocations() {
+    infoWindow.close();
+    for (var i = 0; i < markers.length; i++) {
```

Comment:
again you're using var.
If you don't want to remember those you can use `forEach` array method.